File: backend/fastapi/app/models/article.py
from datetime import datetime, timezone
from typing import Optional
from pydantic import BaseModel, Field

class ArticleBase(BaseModel):
    author: str
    authorId: Optional[str] = None # Link to UserProfile.userId
    title: str
    description: str
    urlToImage: str
    content: str
    publishedAt: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    category: str = "general"

class ArticleCreate(ArticleBase):
    """Schema for creating a new article via API. 
    Only user-contributed fields are present here.
    """
    source: str = "Symmetry User"
    sourceId: Optional[str] = None
    url: Optional[str] = None # Optional, system can generate one if missing

class Article(ArticleBase):
    """Full article model including system-managed fields."""
    articleId: str
    url: str
    source: str
    sourceId: Optional[str] = None
    views: int = 0
    readTime: int = 0
    tokensEarned: float = 0.0
    expectedReadTime: Optional[int] = 30
    qualityScore: Optional[float] = 1.0
    fraudScore: float = 0.0
    rewardEpoch: Optional[int] = 0
    verifiedImpressionCount: int = 0
    pdfPath: Optional[str] = None


    class Config:
        from_attributes = True

class ArticleRead(BaseModel):
    userId: str
    readTimeSeconds: int
    readAt: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

File: frontend/lib/features/daily_news/domain/entities/article.dart
import 'package:equatable/equatable.dart';

class ArticleEntity extends Equatable{
  final String ? articleId;
  final String ? author;
  final String ? authorId;
  final String ? title;
  final String ? description;
  final String ? url;
  final String ? urlToImage;
  final String ? publishedAt;
  final String ? content;
  final double ? tokensEarned;
  final String ? source;
  final String ? pdfPath;
  final int ? upvotes;
  final int ? downvotes;

  const ArticleEntity({
    this.articleId,
    this.author,
    this.authorId,
    this.title,
    this.description,
    this.url,
    this.urlToImage,
    this.publishedAt,
    this.content,
    this.tokensEarned,
    this.source,
    this.pdfPath,
    this.upvotes,
    this.downvotes,
  });


  @override
  List < Object ? > get props {
    return [
      articleId,
      author,
      authorId,
      title,
      description,
      url,
      urlToImage,
      publishedAt,
      content,
      tokensEarned,
      source,
      pdfPath,
      upvotes,
      downvotes,
    ];
  }
}